Contributing.md

```diff
@@ -24,35 +24,36 @@ This guide is for developers who wish to contribute to the Dolphin codebase. It
 
 Following this guide and formatting your code as detailed will likely get your pull request merged much faster than if you don't (assuming the written code has no mistakes in itself).
 
+This project uses clang-format (stable branch) to check for common style issues. In case of conflicts between this guide and clang-format rules, the latter should be followed instead of this guide.
+
+
+### Checking and fixing formatting issues
+
+In most cases, clang-format can and **should** be used to automatically reformat code and solve most formatting issues.
+
+- To run clang-format on all staged files:
+  ```
+  git diff --cached --name-only | egrep '[.](cpp|h|mm)$' | xargs clang-format -style=file -i
+  ```
+
+- Formatting issues can be checked for before committing with a lint script that is included with the codebase. To enable it as a pre-commit hook (assuming you are in the repository root):
+  ```
+  ln -s ../../Tools/lint.sh .git/hooks/pre-commit
+  ```
+
+  To also automatically fix common formatting issues, use the reformat.sh hook instead:
+  ```
+  ln -sf ../../Tools/reformat.sh .git/hooks/pre-commit
+  ```
+  Note that the hook will completely reformat all staged files and commit them, so avoid using git commit -p or add -p or similar when using this hook.
```

Comment:
Note that clang-format supports --lines / --offset to partially reformat a file. Maybe this could be used to only reformat the index.